atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleNamespaceDeleterTests.java

```diff
@@ -0,0 +1,363 @@
+        assertThat(sqliteOracleAdapter.listAllTablesWithMapping())
+                .containsExactlyInAnyOrderElementsOf(TableDetails.mapToPrefixedTableNames(tableDetails));
+        assertThat(sqliteOracleAdapter.listAllTablesWithMetadata())
+                .containsExactlyInAnyOrderElementsOf(TableDetails.mapToTableReferences(tableDetails));
+    }
+
+    private Set<String> listAllPhysicalTableNames() {
+        return sqliteOracleAdapter.listAllPhysicalTableNames();
+    }
+
+    private NamespaceDeleter createDefaultNamespaceDeleter() {
+        return new OracleNamespaceDeleter(
+                createDefaultNamespaceDeleterParameters().build());
+    }
+
+    private ImmutableOracleDdlConfig.Builder getDefaultDdlConfig() {
+        return ImmutableOracleDdlConfig.builder()
+                .tablePrefix(TABLE_PREFIX)
+                .overflowTablePrefix(OVERFLOW_TABLE_PREFIX)
+                .overflowMigrationState(OverflowMigrationState.UNSTARTED)
+                .useTableMapping(true)
+                .metadataTable(SqliteOracleAdapter.METADATA_TABLE);
+    }
+
+    private ImmutableOracleNamespaceDeleterParameters.Builder createDefaultNamespaceDeleterParameters() {
+        return createNamespaceDeleterParameters(getDefaultDdlConfig().build());
+    }
+
+    private ImmutableOracleNamespaceDeleterParameters.Builder createNamespaceDeleterParameters(
+            OracleDdlConfig ddlConfig) {
+        OracleTableNameGetter tableNameGetter = OracleTableNameGetterImpl.createDefault(ddlConfig);
+        ConnectionSupplier connectionSupplier =
+                new ConnectionSupplier(sqliteOracleAdapter.createSqlConnectionSupplier());
+        Function<TableReference, OracleDdlTable> ddlTableFactory = tableReference -> OracleDdlTable.create(
+                tableReference,
+                connectionSupplier,
+                ddlConfig,
+                tableNameGetter,
+                new TableValueStyleCache(),
+                executorService);
+        return OracleNamespaceDeleterParameters.builder()
+                .tablePrefix(ddlConfig.tablePrefix())
+                .overflowTablePrefix(ddlConfig.overflowTablePrefix())
+                .userId(TEST_USER)
+                .oracleDdlTableFactory(ddlTableFactory)
+                .connectionSupplier(connectionSupplier)
+                .tableNameGetter(tableNameGetter);
+    }
+
+    private OracleNamespaceDeleterParameters createNamespaceDeleterParametersWithUnstableFactory() {
+        OracleNamespaceDeleterParameters parameters =
+                createDefaultNamespaceDeleterParameters().build();
+        return OracleNamespaceDeleterParameters.builder()
+                .from(parameters)
+                .oracleDdlTableFactory(new UnstableOracleDdlFactory(parameters.oracleDdlTableFactory()))
+                .build();
+    }
+
+    private static class UnstableOracleDdlFactory implements Function<TableReference, OracleDdlTable> {
```

Comment:
I had an unstable connection supplier but that involved more magic numbers, so this seemed a better way to induce failures in the middle of deleting a bunch of tables whilst ensuring we can still test retries.

happy to take an alternative!